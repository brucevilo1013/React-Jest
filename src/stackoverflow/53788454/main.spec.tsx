import React from 'react';
import { shallow } from 'enzyme';
import Main from './main';
import fetch from 'node-fetch';

const { Response } = jest.requireActual('node-fetch');

jest.mock('node-fetch');

describe('Main', () => {
  it('t1', done => {
    (fetch as jest.MockedFunction<typeof fetch>).mockResolvedValueOnce(
      new Response(
        JSON.stringify({
          data: ['data1', 'data2']
        })
      )
    );
    const wrapper = shallow(<Main></Main>);
    expect(fetch).toBeCalledWith(`http://testurl.com/testData`);
    expect(wrapper).toMatchSnapshot();
    setImmediate(() => {
      expect(wrapper.state()).toEqual({
        testData: {
          data: ['data1', 'data2']
        }
      });
      expect(wrapper).toMatchSnapshot();
      done();
    });
  });
});
