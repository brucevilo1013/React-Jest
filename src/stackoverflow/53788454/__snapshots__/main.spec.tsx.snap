// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main t1 1`] = `
<div
  className="main"
>
  <section>
    <CustomComponent
      movies={Array []}
    />
  </section>
</div>
`;

exports[`Main t1 2`] = `
<div
  className="main"
>
  <section>
    <CustomComponent
      movies={
        Array [
          "data1",
          "data2",
        ]
      }
    />
  </section>
</div>
`;
